### PR TITLE
ARIA, Better Design, Fix Migration, Prettier fmt

### PR DESCRIPTION
**Pull request by FussballAndy**
- Finished migration to SvelteKit
- Added Topics to documentation page
- Redesigned many parts
- Accessibility
- Prettier fmt

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -1,36 +1,38 @@
...
+		"prettier-plugin-svelte": "^2.2.0",
+		"svelte": "^3.34.0",
+		"svelte-preprocess": "^4.0.0",
+		"tailwindcss": "^2.1.2",
+		"tslib": "^2.0.0",
+		"typescript": "^4.0.0",
+		"vite-plugin-svelte-svg": "^1.0.0"
+	},
+	"type": "module",
+	"dependencies": {
```
Comment (Defman) [inline frontend/package.json:32]: I think they should all be dev, also we might want to use tailwindcss-forms? Also maybe upgrade our tailwindcss to jit and use https://github.com/svelte-add/tailwindcss

File: frontend/src/global.css [hunk 2]
```diff
@@ -1,124 +1,118 @@
...
-html, body {
-    @apply h-full flex flex-col;
+html,
+body {
+	@apply h-full flex flex-col;
 }
 
 .typography a {
-    @apply border-b-2 border-feather-accent font-bold;
+	@apply border-b-2 border-feather-accent font-bold;
```
Comment (Defman) [inline frontend/src/global.css:12]: Indent 4 spaces?

File: frontend/src/global.d.ts [hunk 3]
```diff
@@ -1 +1,5 @@
 /// <reference types="@sveltejs/kit" />
+
+declare module '*.svg?component' {
```
Comment (Defman) [inline frontend/src/global.d.ts:3]: Please explain, does this allow us to import svg files as svelte components?

File: frontend/src/lib/components/Card.svelte [hunk 4]
```diff
@@ -1,7 +1,11 @@
...
+	let clazz = undefined;
+	export { clazz as class };
 </script>
-<div class="p-4 shadow border-l-4 border-feather-light {clazz || ""}">
-    <slot></slot>
-</div>
\ No newline at end of file
+
+<div
+	class="p-6 shadow rounded-md bg-gray-100 dark:bg-gray-800 transition-colors duration-200 {clazz ||
```
Comment (Defman) [inline frontend/src/lib/components/Card.svelte:7]: `$$restProps` or `$$props`

File: frontend/src/lib/components/PluginCard.svelte [hunk 5]
```diff
@@ -1,52 +1,77 @@
...
+					{downloads_recent.toLocaleString()}</span
+				>
+			</div>
+			<div class="my-1">
+				<span
+					><abbr title="The last time plugin was updated">Updated:</abbr>
+					{updated.toLocaleString()} days ago</span
+				>
+			</div>
+		</div>
```
Comment (Defman) [inline frontend/src/lib/components/PluginCard.svelte:75]: Indentation, 4 spaces?

File: frontend/tailwind.config.cjs [hunk 6]
```diff
@@ -1,104 +1,167 @@
-const { tailwindExtractor } = require("tailwindcss/lib/lib/purgeUnusedStyles");
+const { tailwindExtractor } = require('tailwindcss/lib/lib/purgeUnusedStyles');
 
 module.exports = {
-  darkMode: "class",
-  mode: "aot",
+	darkMode: 'class',
```
Comment (Defman) [inline frontend/tailwind.config.cjs:4]: I personally prefer `"` over `'`.

**Review by Defman — CHANGES_REQUESTED** [review]
6 or more spaces indentation?

**Comment by FussballAndy** [comment]
The identation was because I ran prettier on it I don't know why it's a bit inconsistent but it should be idented using tabs

File: frontend/src/global.d.ts (hunk 3 above)
Comment (FussballAndy) [reply frontend/src/global.d.ts:3]: Yes

File: frontend/src/lib/components/Card.svelte (hunk 4 above)
Comment (FussballAndy) [reply frontend/src/lib/components/Card.svelte:7]: what do you mean by that?

Comment (Defman) [reply frontend/src/lib/components/Card.svelte:7]: Reference to `clazz` I'm not sure if we are allowed to have more than one `class` though.

Comment (Defman) [reply frontend/src/lib/components/Card.svelte:7]: Also, it might be me who made this...

**Pull request merged**